src/commands.cc

```diff
@@ -1448,22 +1448,32 @@ const CommandDesc declare_option_cmd = {
     }
 };
 
-auto map_key_completer =
+template<bool unmap>
+static auto map_key_completer =
     [](const Context& context, CompletionFlags flags,
        CommandParameters params, size_t token_to_complete,
        ByteCount pos_in_token) -> Completions
     {
         if (token_to_complete == 0)
             return { 0_byte, params[0].length(),
                      complete(params[0], pos_in_token, scopes) };
-
         if (token_to_complete == 1)
         {
-            static constexpr auto modes = { "normal", "insert", "menu", "prompt", "goto", "view", "user", "object" };
             auto& user_modes = get_scope(params[0], context).keymaps().user_modes();
             return { 0_byte, params[1].length(),
                      complete(params[1], pos_in_token, concatenated(modes, user_modes) | gather<Vector<String>>()) };
         }
+        if (unmap && token_to_complete == 2)
```

Comment:
`else if` here?

Reply:
I dont mind much about the `else if`, as the previous if returns from the function, however Kakoune code uses the alternate boolean operator syntax (and, or, not) rather than the "classic" one (&&, ||, !), please fix that.